Add more browser testing with Sauce Labs.

heathkit opened the pull request:
Here's a simple example of one way to add support for testing on Sauce Labs. No need to actually merge this, it's just intended as an example.

heathkit committed:
Setting up Travis with saucelabs.

heathkit committed:
Add a travis.yml with local and saucelabs testing.

heathkit committed:
Trying out a basic port of ci-lite.

heathkit committed:
Trying a simpler way, using Travis's integration.

heathkit committed:
Remember to update webdriver.

heathkit committed:
Only need one Safari.

cnishina commented on tour-of-heroes/protractor-saucelabs.conf.js:
```diff
@@ -0,0 +1,48 @@
+  sauceUser: process.env.SAUCE_USERNAME,
+  sauceKey: process.env.SAUCE_ACCESS_KEY,
+
+  directConnect: false,
```
Instead of trying to override an existing with an Object.assign, could we instead do something like:

```
var defaultConfig = require('./protractor.conf.js').config;

defaultConfig.directConnect = undefined;
defaultConfig.seleniumAddress = undefined;
defaultConfig.sauceUser
...

exports.config = defaultConfig;
```

similar to https://github.com/angular/protractor/blob/6a667c8a3211ac451a264544bb6f44dfe29f7b74/spec/ciNg2Conf.js

cnishina replied:
Also, `defaultConfig.capabilities = undefined` to get rid of the warning

heathkit replied:
Turns out the problem was I was using an old Sauce Labs account, which didn't have any VM minutes for automated testing. Instead of throwing an error, Sauce would just hang forever waiting to launch.

Also, this uncovered a bug in Protractor. The way 'beforeLaunch' is used to set up ts-node doesn't actually work with multiCapabilities, because of the way we fork in the launcher. I suspect that if you used test sharding, that would also break.

heathkit committed:
Use a different sauce account, only use a single browser.

There seems to be an issue with 'beforeLaunch' and multicapabilities that breaks ts-node when
launching.
